controllers/volumereplicationgroup_controller.go

```diff
@@ -1145,6 +1184,74 @@ func (v *VRGInstance) createVR(vrNamespacedName types.NamespacedName, state volr
 	return nil
 }
 
+// namespacedName applies to both VolumeReplication resource and pvc as of now.
+// This is because, VolumeReplication resource for a pvc that is created by the
+// VolumeReplicationGroup has the same name as pvc. But in future if it changes
+// functions to be changed would be processVRAsPrimary(), processVRAsSecondary()
+// to either receive pvc NamespacedName or pvc itself as an additional argument.
+func (v *VRGInstance) selectVolumeReplicationClass(namespacedName types.NamespacedName) (string, error) {
+	className := ""
+
+	if len(v.replClassList.Items) == 0 {
+		v.log.Info("No VolumeReplicationClass available")
+
+		return className, fmt.Errorf("no VolumeReplicationClass available")
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := v.reconciler.Get(v.ctx, namespacedName, pvc); err != nil {
+		v.log.Info("failed to get the pvc with namespaced name", namespacedName)
+
+		// Need the storage driver of pvc. If pvc is not found return error.
+		return className, fmt.Errorf("failed to get the pvc with namespaced name %s", namespacedName)
+	}
+
+	scName := pvc.Spec.StorageClassName
+
+	storageClass := &storagev1.StorageClass{}
+	if err := v.reconciler.Get(v.ctx, types.NamespacedName{Name: *scName}, storageClass); err != nil {
+		v.log.Info(fmt.Sprintf("Failed to get the storageclass %s", *scName))
+
+		return className, fmt.Errorf("failed to get the storageclass with name %s",
+			*scName)
+	}
+
+	for index := range v.replClassList.Items {
+		replicationClass := &v.replClassList.Items[index]
+		if storageClass.Provisioner != replicationClass.Spec.Provisioner {
+			continue
+		}
+
+		schedule, found := replicationClass.Spec.Parameters["schedule"]
+		if !found {
+			// As of now, the provisioner matches. Use this as the best guess
+			// until a replicationClass with schedule match is found. i.e.
```

Comment:
What if you never find a schedule but a provisioner matched, is the className assignment here still valid?

Reply:
Ack! If we choose this we will exit with a non-empty classname that does not match the schedule, let's match both driver name and schedule to select the volReplClass.